nuxeo-runtime/nuxeo-connect-standalone/src/test/java/org/nuxeo/launcher/connect/TestConnectBroker.java

```diff
@@ -284,10 +302,55 @@ public void testDownloadUnknownPackage() throws Exception {
 
         // Assertions
         checkPackagesState(connectBroker, Arrays.asList("unknown-package"), NON_EXISTENT);
-        List<String> expectedLogs = new ArrayList<>();
-        expectedLogs.add("Downloading [unknown-package]...");
-        expectedLogs.add("\tDownload failed (not found).");
-        checkLogEvents(expectedLogs, logCaptureResult.getCaughtEvents());
+        connectBroker.getCommandSet().log();
+        String expectedLogs = "Downloading [unknown-package]...\n" //
+                + "\tDownload failed (not found)."; //
+        assertThat(logOf(logCaptureResult)).isEqualTo(expectedLogs);
+    }
+
+    /**
+     * Reproduction of the commands in <a href="https://jira.nuxeo.com/browse/NXP-21573">NXP-21573</a>, which should be
```

Comment:
This is good but you should add more tricky cases as this is a very very tricky part of the code (speaking of experience ;) )
I think of some cases like :
- requesting the install of F by id
- trying the simpler case when no relax is needed
- what if the package to replace is a dependency of the requested pkg ?
- what about uninstalling a pkg remotely unavailable ?

There are maybe some more you can think of :D, you can add some new tests methods (better I think) or update the existing tests (more risky) but the more tests there are the better it is !!
By the way, good job with the log assertion simplification, it's much more readable like this ;)